pkg/api/v1/mountpoints3podattachment_types.go

```diff
@@ -0,0 +1,79 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// The following fields are used as matching criteria to determine if a mountpoint s3 pod can be shared by having the same MountpointS3PodAttachment resource:
+const (
+	FieldNodeName                         = "spec.nodeName"
+	FieldPersistentVolumeName             = "spec.persistentVolumeName"
+	FieldVolumeID                         = "spec.volumeID"
+	FieldMountOptions                     = "spec.mountOptions"
+	FieldAuthenticationSource             = "spec.authenticationSource"
+	FieldWorkloadFSGroup                  = "spec.workloadFSGroup"
+	FieldWorkloadServiceAccountName       = "spec.workloadServiceAccountName"
+	FieldWorkloadNamespace                = "spec.workloadNamespace"
+	FieldWorkloadServiceAccountIAMRoleARN = "spec.workloadServiceAccountIAMRoleARN"
+)
+
+// MountpointS3PodAttachmentSpec defines the desired state of MountpointS3PodAttachment.
+type MountpointS3PodAttachmentSpec struct {
+	// Important: Run "make generate" to regenerate code after modifying this file
```

Comment:
Would be nice to enforce this as part of our CI if there is an easy way to do

Reply:
Would be still nice to have a way to detect if this is "dirty" meaning it requires us to run `make generate`, but it's not urgent